remoting/subsystem/src/main/java/org/jboss/as/remoting/RemoteOutboundConnectionService.java

```diff
@@ -125,7 +119,7 @@ public void start(final StartContext context) throws StartException {
             URI finalUri = uri;
             authenticationConfiguration = () -> AUTH_CONFIGURATION_CLIENT.getAuthenticationConfiguration(finalUri, injectedContext);
         } else {
-            final SecurityRealm securityRealm = securityRealmInjectedValue.getOptionalValue();
+            final SecurityRealm securityRealm = securityRealmSupplier != null ? securityRealmSupplier.get() : null;
```

Comment:
likewise, would some `Optional<Supplier<SecurityRealm>>` be more self-explanatory?

Reply:
Depends on developer's taste. But I have to agree with you it would be more self-explanatory.